src/app/system/surveys/surveys.component.html

```diff
@@ -0,0 +1,68 @@
+<div class="container m-b-20" fxLayout="row" fxLayoutAlign="end" fxLayoutGap="20px">
+    <button mat-raised-button color="primary" [routerLink]="['create']">
+      <fa-icon icon="plus"></fa-icon>&nbsp;&nbsp;
+      Create Surveys
+    </button>
+  </div>
+  
+  <div class="container">
+  
+    <div fxLayout="row">
+      <mat-form-field fxFlex>
+        <mat-label>Filter</mat-label>
+        <input matInput (keyup)="applyFilter($event.target.value)">
+      </mat-form-field>
+    </div>
+  
+    <div class="mat-elevation-z8">
+      <table mat-table [dataSource]="dataSource" matSort>
+        <ng-container matColumnDef="key">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header='key'> Key </th>
+            <td mat-cell *matCellDef="let row"> {{row.key}} </td>
+          </ng-container>
+          
+          <!-- Name Column -->
+          <ng-container matColumnDef="name">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header='name'> Name </th>
+            <td mat-cell *matCellDef="let row"> {{row.name}} </td>
+          </ng-container>
+    
+          <!-- Description Column-->
+          <ng-container matColumnDef="description">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header='description'> Description </th>
+            <td mat-cell *matCellDef="let row"> {{row.description}} </td>
+          </ng-container>
+    
+          <!-- CountryCode Column -->
+          <ng-container matColumnDef="countryCode">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header='countryCode'> Country Code </th>
+            <td mat-cell *matCellDef="let row"> {{row.countryCode}} </td>
+          </ng-container>
+    
+    
+          <ng-container matColumnDef="status">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header='status'> Status </th>
+            <td mat-cell *matCellDef="let row">
+                <p mat-raised-button [hidden]="!isActive(row.validFrom, row.validTo)" >Active</p>
+                <p mat-raised-button [hidden]="isActive(row.validFrom, row.validTo)" >Inactive</p>
+            </td>
+          </ng-container>
+    
+          <!-- Status Column -->
+          <ng-container matColumnDef="action">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header='action'> Action </th>
+            <td mat-cell *matCellDef="let row" value="isActive(row.validFrom, row.validTo)">
+                <a mat-raised-button [hidden]="!isActive(row.validFrom, row.validTo)" type="button" style="background: #e53935; color: white;" ng-click="activateOrDeactivateSurvey(row.id,'deactivate')">Deactivate</a>
```

Comment:
Don't use inline styles. Only classes. For these buttons, use `color="accent"` for activate and `color="warn"` for deactivate. No other styles are required here.